### PR TITLE
[fx] added colotensor in 1d linear pass

### PR DESCRIPTION
**Pull request by Itok2000u**
Linear 1D pass with colotensor. Specifically, rewriting weight_split func with tensor_spec property of colotensor.

### CODE REVIEW

**Comment by FrankLeeeee** [comment]
Hi, can you change the PR name to be the format `[<item>] <message>`, e.g. `[fx] added a control-flow tracer`.

File: colossalai/fx/passes/shard_1d_pass.py [hunk 1]
```diff
@@ -26,28 +28,31 @@ def all_reduce_function(input_):
...
-    num_partition = gpc.get_world_size(ParallelMode.TENSOR)
-    shape = weight.shape
-    length = shape[dim] // num_partition
-    sharded_weight_list = []
-    for i in range(num_partition):
-        sharded_weight_list.append(weight.narrow(dim, i * length, length))
-    return sharded_weight_list[gpc.get_local_rank(ParallelMode.PARALLEL_1D)]
+def weight_split(weight):
+    # split tensor with tensor_spec
+    assert isinstance(weight, colossalai.tensor.colo_tensor.ColoTensor), \
```
Comment (FrankLeeeee) [inline colossalai/fx/passes/shard_1d_pass.py:33]: Does this work with the unit test?

Comment (Itok2000u) [reply colossalai/fx/passes/shard_1d_pass.py:33]: Yes, it worked on the tests of the previous version of shard_1d_pass.py

File: colossalai/fx/passes/shard_1d_pass.py [hunk 2]
```diff
@@ -26,28 +28,31 @@ def all_reduce_function(input_):
...
+        f'The type of the input tensor should be colotensor' \
+        f'Your Input tensor is {type(weight)}'
+    # assert weight.device.type == "meta"
+    tensor_dist_spec = weight.tensor_spec
+    num_partition = tensor_dist_spec.get_process_group_size()
```
Comment (feifeibear) [inline colossalai/fx/passes/shard_1d_pass.py:38]: The API has changed!
current it should be
```
tensor_dist_spec.process_group.tp_world_size()
```

Comment (Itok2000u) [reply colossalai/fx/passes/shard_1d_pass.py:38]: Thank you. I will check that!

**Pull request closed without merging**